### PR TITLE
Set ENABLE_TESTING_SEARCH_PATHS to NO

### PR DESCRIPTION
**Pull request by wattson12**
Resolves https://github.com/tuist/tuist/issues/3526
Request for comments document (if applies):

### Short description 📝

This is following the discussion in https://github.com/tuist/tuist/issues/3526 where archive / for-device builds are failing due to bitcode errors when SPM dependencies are added via `Tuist/Dependencies.swift`

This is following the suggestion from @thedavidharris to set `ENABLE_TESTING_SEARCH_PATHS` to `NO`

I believe this was set to true because this is what [SPM does](https://github.com/apple/swift-package-manager/blob/c42b5171cf5dece3bd4922161fb10a63833df794/Sources/XCBuildSupport/PIFBuilder.swift#L283) but if I create an empty project and use `swift package generate-xcodeproj` this value is `NO`

I'm not sure of the broader impact of this change but I was able to archive a project using the `Apollo` library (the library referenced in the original ticket) from this branch

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [x] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

**Comment by danyf90** [comment]
I think this causes problems to SPM packages requiring testing search paths (e.g. [Quick](https://github.com/Quick/Quick)/[Nimble](https://github.com/Quick/Nimble)

The CI should catch the problem as they are used in the SPM fixture, let's see 🤞

**Comment by wattson12** [comment]
> I think this causes problems to SPM packages requiring testing search paths (e.g. [Quick](https://github.com/Quick/Quick)/[Nimble](https://github.com/Quick/Nimble)

Yeah I was wondering what this would break, it looks like they have already failed. Is there anything about these packages which we can use to determine that they require this flag to be set to true? Initially I started setting this to something like 

```
"ENABLE_TESTING_SEARCH_PATHS": target.type == .test 
```

but I think those targets are filtered out

**Comment by danyf90** [comment]
Unfortunately I haven't found how to understand that 🤷 
If we don't find a way, once [this](https://github.com/tuist/tuist/pull/3611) is merged, We might even default to NO, and leave the developers set the YES only where needed.

**Comment by wattson12** [comment]
> Unfortunately I haven't found how to understand that 🤷
> If we don't find a way, once [this](https://github.com/tuist/tuist/pull/3611) is merged, We might even default to NO, and leave the developers set the YES only where needed.

Nice, makes sense. I would guess `NO` is by far the default. I wonder if we could get closer by setting different defaults for sim vs device. I don't know quick / nimble well so I'm assuming you can run on device but sim would be more common?

**Review by fortmarek — APPROVED** [review]
> Nice, makes sense. I would guess NO is by far the default. I wonder if we could get closer by setting different defaults for sim vs device. I don't know quick / nimble well so I'm assuming you can run on device but sim would be more common?

I'm for changing the default, although we should still try to fix this properly in the future.

**Comment by danyf90** [comment]
Good to me to change the default, but I would still wait for https://github.com/tuist/tuist/pull/3611 to be merged to avoid breaking stuff which is working right now.
@thedavidharris do you think you'll have time to bring that PR to the finish line? 🙏

**Comment by wattson12** [comment]
> Good to me to change the default, but I would still wait for #3611 to be merged to avoid breaking stuff which is working right now.
> @thedavidharris do you think you'll have time to bring that PR to the finish line? 🙏

definitely makes sense to wait for that one, probably to update the failing acceptance test to work via those settings

**Comment by ezraberch** [comment]
If you add `Nimble` as a dependency using CocoaPods, it sets this flag though the `.podspec`. If you use Carthage, I think it uses the `.xcodeproj`.

I am not sure what the equivalent is for SPM. `Nimble` does not mention support for SPM, so perhaps there isn't one. If that is the case, how about allowing this to be set in Dependences.swift?

**Comment by danyf90** [comment]
> Nimble does not mention support for SPM, so perhaps there isn't one

What do you mean @ezraberch ? Nimble has its [Package.swift](https://github.com/Quick/Nimble/blob/main/Package.swift) so it supports SPM 🤷

**Comment by thedavidharris** [comment]
> If you add `Nimble` as a dependency using CocoaPods, it sets this flag though the `.podspec`. If you use Carthage, I think it uses the `.xcodeproj`.
> 
> I am not sure what the equivalent is for SPM. `Nimble` does not mention support for SPM, so perhaps there isn't one. If that is the case, I agree with letting the behavior be specified in Dependencies.swift.

There's something weird with how the Swift Packages are built where it picks up the testing search paths as needed, but it's not super easy to find in the build system when I did a quick look through. Generated projects would have this same issue if you used that deprecated command.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 - **Breaking** Update logic to calculate deployment target for SwiftPackageManager packages not specifying it, and remove no longer used `SwiftPackageManagerDependencies.deploymentTargets` property [#3602](https://github.com/tuist/tuist/pull/3602) by [@danyf90](https://github.com/danyf90)
 - **Breaking** Update logic to calculate client ID starting from UUID instead of hostname, to avoid collisions [#3632](https://github.com/tuist/tuist/pull/3632) by [@danyf90](https://github.com/danyf90)
 - `Target`'s initializer now has `InfoPlist.default` set as the default value for the `infoPlist` argument [#3644](https://github.com/tuist/tuist/pull/3644) by [@hisaac](https://github.com/hisaac)
+- **Breaking** Switch default value for `ENABLE_TESTING_SEARCH_PATHS` in SPM dependencies to `NO` (workaround for targets requiring XCTest provided by [#3611](https://github.com/tuist/tuist/pull/3611)) [#3632](https://github.com/tuist/tuist/pull/3653) by [@wattson12](https://github.com/wattson12)
```
Comment (danyf90) [inline CHANGELOG.md:12]: Also, let's put it together with the other breakings above
```suggestion
- **Breaking** Switch default value for `ENABLE_TESTING_SEARCH_PATHS` in SPM dependencies to `NO`. If a target requires it, you can set it using the `targetSettings` property in the `Dependencies.swift` file [#3632](https://github.com/tuist/tuist/pull/3653) by [@wattson12](https://github.com/wattson12)
```

**Review by danyf90 — APPROVED** [review]

**Comment by ezraberch** [comment]
Is it necessary to explicitly set `ENABLE_TESTING_SEARCH_PATHS` to NO? Since the default is NO, not setting it should have the same effect.

> > Nimble does not mention support for SPM, so perhaps there isn't one
> 
> What do you mean @ezraberch ? Nimble has its [Package.swift](https://github.com/Quick/Nimble/blob/main/Package.swift) so it supports SPM 🤷

I meant that the Nimble docs don't list it as one of the supported methods. (Quick lists SPM support as experimental.)

**Comment by danyf90** [comment]
> I meant that the Nimble docs don't list it as one of the supported methods. (Quick lists SPM support as experimental.)
I see, but it somehow works when using SPM 🤷 

> Is it necessary to explicitly set ENABLE_TESTING_SEARCH_PATHS to NO? Since the default is NO, not setting it should have the same effect.
I agree with @ezraberch , @wattson12 could you please fix it? 🙏

**Comment by wattson12** [comment]
> > Is it necessary to explicitly set ENABLE_TESTING_SEARCH_PATHS to NO? Since the default is NO, not setting it should have the same effect.

Makes sense 👍 pushed

**Review by danyf90 — APPROVED** [review]

**Pull request merged**